### PR TITLE
Internal sharing

### PR DESCRIPTION
**Pull request by mjoyce91**
This PR addresses #119 - Share a position with another employee. A radio button is now displayed above the email input to allow the user to choose between an Internal and an External share. Additionally, if there is an error message from the server, such as "User does not exist", this is displayed to the user instead of a generic error message.

### CODE REVIEW

File: src/Components/Share/ShareButton/ShareButton.jsx [hunk 1]
```diff
@@ -29,40 +30,70 @@ class ShareButton extends Component {
...
+            />
+            <label htmlFor="internal">Internally</label>
+          </fieldset>
+          <input
+            id="share-input"
+            name="input-type-text"
+            type="text"
+            value={this.state.recipient}
+            onChange={e => this.changeEmail(e)}
+            placeholder="Receipient's email address"
```
Comment (burgwyn) [inline src/Components/Share/ShareButton/ShareButton.jsx]: Typo

File: src/Components/Share/ShareButton/ShareButton.jsx [hunk 2]
```diff
@@ -29,40 +30,70 @@ class ShareButton extends Component {
...
+          <button className={(recipient.length && !isSending) ? null : 'usa-button-disabled'} disabled={recipient.length ? null : true} id="share-button">
           Share
         </button>
-        {warning && recipient.length ? 'This is not a state.gov email. Send with caution.' : null}
-        <br />
-        {sendingText} {err} {sent}
-      </form>
+          {warning && recipient.length ? 'This is not a state.gov email. Send with caution.' : null}
+          <br />
+          {sendingText} {err} {sent}
```
Comment (burgwyn) [inline src/Components/Share/ShareButton/ShareButton.jsx]: Could all three or more than one have values simultaneously?

File: src/Components/Share/ShareButton/ShareButton.jsx [hunk 3]
```diff
@@ -29,40 +30,70 @@ class ShareButton extends Component {
...
+          </fieldset>
+          <input
+            id="share-input"
+            name="input-type-text"
+            type="text"
+            value={this.state.recipient}
+            onChange={e => this.changeEmail(e)}
+            placeholder="Receipient's email address"
+          />
+          <button className={(recipient.length && !isSending) ? null : 'usa-button-disabled'} disabled={recipient.length ? null : true} id="share-button">
```
Comment (burgwyn) [inline src/Components/Share/ShareButton/ShareButton.jsx]: I'm not sure about using `recipient.length` in validations.  Could it throw false positives?  Could we flag some of this inside `share(e)`?

File: src/Components/Share/ShareButton/ShareButton.jsx (hunk 1 above)
Comment (mjoyce91) [inline src/Components/Share/ShareButton/ShareButton.jsx]: 🔠

File: src/Components/Share/ShareButton/ShareButton.jsx (hunk 3 above)
Comment (mjoyce91) [inline src/Components/Share/ShareButton/ShareButton.jsx]: All I'm really checking for here is that _some_ text has been entered and a share is not currently being sent -- if so, the button is disabled. I think what @jseppi and I discussed was that a lot more validation will ultimately take place on the server, and we can adjust the front-end accordingly when that happens.

File: src/Components/Share/ShareButton/ShareButton.jsx (hunk 2 above)
Comment (mjoyce91) [inline src/Components/Share/ShareButton/ShareButton.jsx]: You're right - that is possible. I updated the action to set the state of `hasErrored`, `isSending`, and `Success` in such a way that only one of the three is ever true.

**Pull request merged**